docs/Using_Benchmark.md

```diff
@@ -0,0 +1,285 @@
+
+We can add Benchmark as a submodule in our CMake project, build it and use it 
+in the project. To demonstrate this method we consider a simple project 
+`BenchmarkEverything` whose directory structure is as follows:
+
+```bash
+BenchmarkEverything
+|   CMakeLists.txt
+|   BenchmarkEverything.cpp
+|   .git
+```
+
+We add Benchmark as a submodule to our git repository using
+```bash
+$ git submodule add http://github.com/google/benchmark.git extproject/benchmark
+```
+This will add Benchmark as a submodule in `/extproject/benchmark/`. In our 
+`CMakeLists.txt` file we incorporate this submodule into our project using `add_subdirectory()`
+```cmake
+# setting minimum cmake version
+cmake_minimum_required (VERSION 3.14)
+# setting cxx standard
+set (CMAKE_CXX_STANDARD 11)
+set (CMAKE_CXX_STANDARD_REQUIRED ON)
+# project name
+project (BenchmarkEverything)
+# disabling tests for benchmark
+set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Disable benchmark tests" FORCE)
+add_subdirectory(extprojects/benchmark)
+# including benchmark header files
+include_directories(extprojects/benchmark/include)
+add_executable(BenchmarkEverything BenchmarkEverything.cpp)
+target_link_libraries(BenchmarkEverything benchmark)
+if(WIN32)
+    target_link_libraries(BenchmarkEverything shlwapi)
+elseif(LINUX)
+    target_link_libraries(BenchmarkEverything pthread)
+elseif(SOLARIS)
+    target_link_libraries(BenchmarkEverything kstat)
+endif()
+```
+Here we have disabled the build and run of tests for Benchmark by 
+setting `BENCHMARK_ENABLE_TESTING` to `OFF`. If we wish to build and run the 
+tests we can either include googletest as a submodule or set 
+`BENCHMARK_DOWNLOAD_DEPENDENCIES` to `ON`. In both cases we need to remove the 
+line setting `BENCHMARK_ENABLE_TESTING` to `OFF`.
+
+In the file `BenchmarkEverything.cpp` we add
+```c++
+// An simple example from 
+// https://github.com/google/benchmark/blob/master/README.md
+#include <benchmark/benchmark.h>
+
+static void BM_StringCreation(benchmark::State& state) {
+  for (auto _ : state)
+    std::string empty_string;
+}
+// Register the function as a benchmark
+BENCHMARK(BM_StringCreation);
+
+// Define another benchmark
+static void BM_StringCopy(benchmark::State& state) {
+  std::string x = "hello";
+  for (auto _ : state)
+    std::string copy(x);
+}
+BENCHMARK(BM_StringCopy);
+
+BENCHMARK_MAIN();
+```
+
+We can then then build `BenchmarkEverything` from `BenchmarkEverything/` 
+using CMake as follows:
+```shell
+$ cmake -S . -B <BUILD_DIR> -G <GENERATOR>
+$ cmake --build <BUILD_DIR> --config <CONFIGURATION>
+```
+where `<BUILD_DIR>` is the directory where you would like to build 
+`BenchmarkEverything`, `<GENERATOR>` is the generator that you would 
+like to use and `<CONFIGURATION>` the configuration that you would like 
+to build, like `Release` or `Debug`. To run `BenchmarkEverything` we just 
+need to run the executable generated inside `<BUILD_DIR>`.
+
+## Use CMake for embedding Benchmark
```

Comment:
There is one more variant - download it at cmake time, and then fall-back into `Adding Benchmark as a git submodule` variant.

Reply:
that would require you understand both given options first. someone doing that probably is an advanced coder